metagraph/src/annotation/binary_matrix/row_sparse_disk/row_sparse_disk.hpp

```diff
@@ -0,0 +1,107 @@
+        return Impl(boundary_, int_vector_buffer_params.filename,
+                    int_vector_buffer_params.offset, int_vector_buffer_params.buff_size)
+                .get_row(row);
+    }
+    std::vector<Row> get_column(Column column) const override {
+        return Impl(boundary_, int_vector_buffer_params.filename,
+                    int_vector_buffer_params.offset, int_vector_buffer_params.buff_size)
+                .get_column(num_rows(), column);
+    }
+
+    std::vector<SetBitPositions> get_rows(const std::vector<Row> &rows) const override {
+        return Impl(boundary_, int_vector_buffer_params.filename,
+                    int_vector_buffer_params.offset, int_vector_buffer_params.buff_size)
+                .get_rows(rows);
+    }
+
+    bool load(std::istream &in) override;
+    void serialize(std::ostream &out) const override;
+
+    // number of ones in the matrix
+    uint64_t num_relations() const override {
+        return Impl(boundary_, int_vector_buffer_params.filename,
+                    int_vector_buffer_params.offset, int_vector_buffer_params.buff_size)
+                .num_relations_impl();
+    }
+
+    static void serialize(const std::function<void(binmat::BinaryMatrix::RowCallback)> &call_rows,
+                          const std::string &filename,
+                          uint64_t num_cols,
+                          uint64_t num_set_bits,
+                          uint64_t num_rows);
+
+    void set_buff_size(uint64_t buff_size) {
+        int_vector_buffer_params.buff_size = buff_size;
+    }
+
+    const bit_vector_small get_boundary() const { return boundary_; }
+
+  private:
+    // For the multithreading to work properly the idea is to open int_vector_buffer<> per
+    // each public method call that needs to work with int_vector_buffer<> The actual
+    // implementation is wrapped in the class below to assure it will not call any methods
+    // of RowSparseDisk class which could result in multiple opens of int_vector_buffer
+    class Impl {
+      public:
+        Impl(const bit_vector_small &boundary,
+             const std::string &filename,
+             uint64_t offset,
+             uint64_t buff_size)
+            : boundary_(boundary),
+              set_bits_(filename, std::ios::in, buff_size, 0, false, offset) {}
+
+        uint64_t num_relations_impl() const { return set_bits_.size(); }
+        bool get(Row row, Column column) const;
+        BinaryMatrix::SetBitPositions get_row(Row row) const;
+        std::vector<SetBitPositions> get_rows(const std::vector<Row> &row_ids) const;
+        std::vector<BinaryMatrix::Row> get_column(uint64_t num_rows, Column column) const;
+
+      private:
+        const bit_vector_small &boundary_;
+        sdsl::int_vector_buffer<> set_bits_;
+    };
+
+    struct {
+        std::string filename;
+        uint64_t offset;
+        uint64_t buff_size = 10'000;
+    } int_vector_buffer_params;
```

Comment:
```suggestion
    struct int_vector_buffer_params {
        std::string filename;
        uint64_t offset;
        uint64_t buff_size = 10'000;
    };
```

Reply:
This wouldn't compile, I mean I will need to create an instance of this struct. My idea was to just have an instance of an unnamed type since the type is not needed elsewhere. The idea was to just have a single instance keeping `int_vector_buffer` parameters together. Anyway, if this is somehow against metagraph's code style I will change it into:
```
struct int_vector_buffer_params_t {
        std::string filename;
        uint64_t offset;
        uint64_t buff_size = 10'000;
    };
int_vector_buffer_params_t int_vector_buffer_params;
```
Let me know.

Reply:
Why would it not compile?

Reply:
Fixed